pinot-core/src/test/java/org/apache/pinot/queries/ExplainPlanQueriesTest.java

```diff
@@ -0,0 +1,510 @@
+    result1.add(new Object[]{"FILTER_TEXT_INDEX(indexLookUp:text_index,operator:TEXT_MATCH,predicate:text_match"
+        + "(textIndexCol1,'foo'))", 5, 4});
+    check(query1, new ResultTable(DATA_SCHEMA, result1));
+
+    String query2 =
+        "EXPLAIN PLAN FOR SELECT noIndexCol1, max(noIndexCol2), min(noIndexCol3) FROM testTable WHERE TEXT_MATCH"
+            + "(textIndexCol1, 'foo') GROUP BY noIndexCol1, noIndexCol2 ORDER BY noIndexCol1, max(noIndexCol2)";
+    List<Object[]> result2 = new ArrayList<>();
+    result2.add(new Object[]{"BROKER_REDUCE(sort:[noIndexCol1 ASC, max(noIndexCol2) ASC],limit:10)", 0, -1});
+    result2.add(new Object[]{"COMBINE_GROUPBY_ORDERBY", 1, 0});
+    result2.add(new Object[]{
+        "AGGREGATE_GROUPBY_ORDERBY(groupKeys:noIndexCol1,noIndexCol2,aggregations:max(noIndexCol2), min"
+            + "(noIndexCol3))", 2, 1});
+    result2.add(new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:noIndexCol1, invertedIndexCol1)", 3, 2});
+    result2.add(new Object[]{"PROJECT(noIndexCol3, noIndexCol2, noIndexCol1)", 4, 3});
+    result2.add(new Object[]{"FILTER_TEXT_INDEX(indexLookUp:text_index,operator:TEXT_MATCH,predicate:text_match"
+        + "(textIndexCol1,'foo'))", 5, 4});
+  }
+
+  @Test
+  public void testSelectAggregate() {
+    String query1 = "EXPLAIN PLAN FOR SELECT count(*) FROM testTable";
+    List<Object[]> result1 = new ArrayList<>();
+    result1.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result1.add(new Object[]{"COMBINE_AGGREGATE", 1, 0});
+    result1.add(new Object[]{"AGGREGATE_METADATA", 2, 1});
+    check(query1, new ResultTable(DATA_SCHEMA, result1));
+
+    String query2 =
+        "EXPLAIN PLAN FOR SELECT count(*), max(noIndexCol1), sum(noIndexCol2), avg(noIndexCol2) FROM testTable";
+    List<Object[]> result2 = new ArrayList<>();
+    result2.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result2.add(new Object[]{"COMBINE_AGGREGATE", 1, 0});
+    result2.add(
+        new Object[]{"AGGREGATE(aggregations:count(*), max(noIndexCol1), sum(noIndexCol2), avg(noIndexCol2))", 2, 1});
+    result2.add(new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:noIndexCol1, noIndexCol2)", 3, 2});
+    result2.add(new Object[]{"PROJECT(noIndexCol2, noIndexCol1)", 4, 3});
+    result2.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 5, 4});
+    check(query2, new ResultTable(DATA_SCHEMA, result2));
+  }
+
+  @Test
+  public void testSelectAggregateUsingFilter() {
+    String query = "EXPLAIN PLAN FOR SELECT sum(add(noIndexCol1, noIndexCol2)), MIN(ADD(DIV(noIndexCol1,noIndexCol2),"
+        + "noIndexCol3)) FROM testTable";
+    List<Object[]> result = new ArrayList<>();
+    result.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result.add(new Object[]{"COMBINE_AGGREGATE", 1, 0});
+    result.add(new Object[]{
+        "AGGREGATE(aggregations:sum(add(noIndexCol1,noIndexCol2)), min(add(div(noIndexCol1,noIndexCol2),"
```

Comment:
The plan is not correct. Since aggregation is happening on top of a transform, we should reflect that in the plan

Reply:
TRANSFORM operator was getting ignored in EXPLAIN PLAN output. Added it back.